content/news/2021/2021-09-01-announcing-the-new-exceptionless-javascript-client.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Announcing The New Exceptionless JavaScript Client
+date: 2021-09-03
+draft: false
+---
+    
+We have had a JavaScript client for several years, but it was quickly becoming out of date. To help address the issues with a fast-moving JavaScript landscape and to extend our JavaScript support, we have recently a completely rewritten client that supports multiple JavaScript frameworks. 
```

Comment:
Our first client release was in 2015 (https://github.com/exceptionless/Exceptionless.JavaScript/releases/tag/v0.1.0). I wouldn't necessarily say it was out of date, those clients will continue to work and be useful probably for another decade or longer. It just was built before ES6 and before the major browsers supported everything. We built it and could only depend on callbacks and ES3 feature set.  We may want to say this. Our main goals were to modernize the api into what would be idiomatic JavaScript code today. Making use of async and await, fetch and ESM modules (tree shaking). Making the api surface feel like code you would right today. By splitting it out into packages we could build smaller libraries and only consume what you need while allowing us to target specific frameworks.

Reply:
@niemyjski we are talking about new stuff. We need to sensationalize that as much as possible. I do think we should mention the facts that we are built on TypeScript with strong typing and using async await and fetch. I think it would also be nice to list out the architectural structure where we have a core client that just uses fetch and localstorage for all features. Then we have layers on top of that the provide platform specific things like wiring into different error handlers and having to provide polyfills for things like fetch in nodejs.

Reply:
I also think it's important that we have at least one image in these articles not only to break things up, but for showing in social media where having an image saying what the tweet is about instead of just text makes a big different for visual scanning through a list of tweets.

Reply:
![image](https://user-images.githubusercontent.com/282584/132272465-5db925bd-85b5-4db8-a315-d067be4ecb06.png)

I like how @actipro does it. https://twitter.com/Actipro/status/1313115200355262464?s=20

Reply:
Agreed. I think playing up the core architecture makes a lot of sense. I'll get an image put together today.

Reply:
Lol @ejsmith are you saying my beautiful non-designer design needs work? We'll keep iterating on it :)

Reply:
Haha, yeah, I know. My attempt would probably be even worse. :-)

Reply:
Ok just updated this. Let me know what you think.